account_index_based_currency/models/account_invoice.py

```diff
@@ -0,0 +1,130 @@
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.depends('date_invoice', 'index_based_currency_id',
+                 'company_currency_id', 'currency_id')
+    def _compute_currency_rate(self):
+        for inv in self:
+            inv.index_based_currency_rate = 1
+            inv.currency_rate = (
+                inv.currency_id._convert(
+                    1, inv.index_based_currency_id, inv.company_id,
+                    inv.date_invoice or fields.Date.today(), round=False))
+            inv.company_currency_rate = (
+                inv.company_currency_id._convert(
+                    1, inv.index_based_currency_id, inv.company_id,
+                    inv.date_invoice or fields.Date.today(), round=False))
+
+    @api.depends('amount_total_signed', 'agreement_currency_rate',
+                 'date_invoice')
+    def _compute_currency_amount(self):
+        for inv in self:
+            index_based_currency_amount = inv.amount_total_signed
+            if inv.index_based_currency_id != inv.currency_id:
+                index_based_currency_amount = inv.currency_id._convert(
+                    inv.amount_total_signed, inv.index_based_currency_id,
+                    inv.company_id, inv.date_invoice or fields.Date.today())
+            inv.index_based_currency_amount = index_based_currency_amount
+            inv.agreement_currency_amount = (
+                inv.index_based_currency_amount / inv.agreement_currency_rate)
+
+    @api.onchange('agreement_currency_rate')
+    def onchange_agreement_currency_rate(self):
+        if self.agreement_currency_rate <= 0:
+            raise UserError(_('Invalid currency rate value'))
+
+    @api.onchange('agreement_currency_id')
+    def onchange_agreement_currency_id(self):
+        self.agreement_currency_rate = self.agreement_currency_id._convert(
+            1, self.index_based_currency_id,
+            self.company_id,
+            self.date_invoice or fields.Date.today(), round=False)
+        self.agreement_currency_amount = self.index_based_currency_id._convert(
+            self.index_based_currency_amount, self.agreement_currency_id,
+            self.company_id,
+            self.date_invoice or fields.Date.today(), round=False)
+
+    @api.model
+    def default_get(self, default_fields):
+        res = super(AccountInvoice, self).default_get(default_fields)
+        default_rate = (
+            self.env.user.company_id.currency_id._convert(
+                1, self.env.user.company_id.index_based_currency_id,
```

Comment:
What about a multi-company user affecting an invoice of a child company?
I mean,

- Company1
  - Company1.1

user.company = Company1
invoice.company = Company1.1